OSDOCS-11435: Network Observability 1.6.1 release notes (#79380)
openshift/openshift-docs: observability/network_observability/network-observability-operator-release-notes.adoc

* Previously, when Loki was disabled, and the OpenShift web console was first loading, an error would occur: `Request failed with status code 400 Loki is disabled`. With this fix, the errors no longer occur. (link:https://issues.redhat.com/browse/NETOBSERV-1706[*NETOBSERV-1706*])
* Previously, in the *Topology* view of the OpenShift web console, when clicking on the *Step into* icon next to any graph node, the filters were not applied as required in order to set the focus to the selected graph node, resulting in showing a wide view of the *Topology*. Now the filters are correctly set, effectively narrowing down the *Topology*. As part of this change, clicking the *Step into* icon on a graph node now brings you to *Resource* scope instead of *Namespaces* scope. (link:https://issues.redhat.com/browse/NETOBSERV-1720[*NETOBSERV-1720*])

> * Previously, when Loki was disabled, in the *Topology* view of the OpenShift web console with the *Scope* set to *Owner*, clicking on the *Step into* icon next to any graph node would bring the *Scope* to *Resource*, which isn't available without Loki, so an error message was shown. Now the *Step into* icon is hidden in the *Owner* scope when Loki is disabled, so this scenario is no longer occurs.(link:https://issues.redhat.com/browse/NETOBSERV-1721[*NETOBSERV-1721*])

Reviewer: extra `is` at the end. Needs to be updated to `so this scenario no longer occurs.`

* When creating a `FlowCollector` resource from the OpenShift web console *Form view*, as opposed to the *YAML view*, the following settings are incorrectly managed by the web console: `agent.ebpf.metrics.enable` and `processor.subnetLabels.openShiftAutoDetect`. These settings can only be disabled in the *YAML view*, not in the *Form view*, . To avoid any confusion, these settings have been removed from the form view. They are still accessible in YAML. (link:https://issues.redhat.com/browse/NETOBSERV-1731[*NETOBSERV-1731*])

> * Previously, the eBPF agent was unable to clean up traffic control flows installed before an ungraceful crash, for example a crash due to a SIGTERM signal. This led to the creation of multiple traffic control flow filters with the same name, since the older ones were not removed. With this fix, all previously installed traffic control flows are cleaned up when the agent starts, before installing new ones.

Author: This JIRA is a Private Jira, which we aren't allowed to paste in the release notes. I originally had the link, but if we link to a private Jira, it breaks our builds.

> * Previously, the eBPF agent was unable to clean up traffic control flows installed before an ungraceful crash, for example a crash due to a SIGTERM signal. This led to the creation of multiple traffic control flow filters with the same name, since the older ones were not removed. With this fix, all previously installed traffic control flows are cleaned up when the agent starts, before installing new ones.

Reviewer: ohh okay

> * Previously, the eBPF agent was unable to clean up traffic control flows installed before an ungraceful crash, for example a crash due to a SIGTERM signal. This led to the creation of multiple traffic control flow filters with the same name, since the older ones were not removed. With this fix, all previously installed traffic control flows are cleaned up when the agent starts, before installing new ones.

Author: I'm going to add a bolded number without the link though, just for consistency. If anyone internal wants to look it up they can.

For an overview of the Network Observability Operator, see xref:../../observability/network_observability/network-observability-overview.adoc#dependency-network-observability[About Network Observability Operator].

[id="network-observability-operator-release-notes-1-6-1_{context}"]
== Network Observability Operator 1.6.1
The following advisory is available for the Network Observability Operator 1.6.1:

* link:https://access.redhat.com/errata/RHSA-2024:4785[2024:4785 Network Observability Operator 1.6.1]

[id="network-observability-operator-CVE-1-6-1_{context}"]
=== CVEs
* link:https://access.redhat.com/errata/RHSA-2024:4237[RHSA-2024:4237]
* link:https://access.redhat.com/errata/RHSA-2024:4212[RHSA-2024:4212]

[id="network-observability-operator-1.6.1-bug-fixes_{context}"]

> === Bug fixes

Reviewer 2: I'm not sure whether using the "with this update" phrasing is something that needs to be used without deviation in Release Notes, but just wanted to point out that seemingly none of the bug fixes here use that exact wording. 

Up to you whether you want to change all instances, but I personally feel like at the very least the fixes that use "Now," phrasing should be changed to be more like the fixes that use "with this fix". That's just my opinion though

Author: Changed all to With this fix

> * Previously, packet drops information, such as cause and TCP state, was only available in the Loki datastore and not in Prometheus. For that reason, the drop statistics in the OpenShift web console plugin *Overview* was only available with Loki. Now, packet drops information is also added to metrics, so you can view drops statistics when Loki is disabled. (link:https://issues.redhat.com/browse/NETOBSERV-1649[*NETOBSERV-1649*])

Reviewer 2: This first sentence is a little confusing at first glance, could it be reworded for more clarity? This suggestion is only based on my interpretation so feel free to change it based on accuracy:
```suggestion
* Previously, information about packet drops, such as the cause and TCP state, was only available in the Loki datastore and not in Prometheus. For that reason, the drop statistics in the OpenShift web console plugin *Overview* was only available with Loki. Now, packet drops information is also added to metrics, so you can view drops statistics when Loki is disabled. (link:https://issues.redhat.com/browse/NETOBSERV-1649[*NETOBSERV-1649*])
```

> * Previously, packet drops information, such as cause and TCP state, was only available in the Loki datastore and not in Prometheus. For that reason, the drop statistics in the OpenShift web console plugin *Overview* was only available with Loki. Now, packet drops information is also added to metrics, so you can view drops statistics when Loki is disabled. (link:https://issues.redhat.com/browse/NETOBSERV-1649[*NETOBSERV-1649*])

Reviewer 2: Similar suggestion about rephrasing in the last sentence:
```suggestion
* Previously, packet drops information, such as cause and TCP state, was only available in the Loki datastore and not in Prometheus. For that reason, the drop statistics in the OpenShift web console plugin *Overview* was only available with Loki. Now, information about packet drops is also added to metrics, so you can view drops statistics when Loki is disabled. (link:https://issues.redhat.com/browse/NETOBSERV-1649[*NETOBSERV-1649*])
```

Although if "packet drops information" is some specific term, then ignore these comments.

> * When the eBPF agent `PacketDrop` feature is enabled, and sampling is configured to a value greater than `1`, reported dropped bytes and dropped packets ignore the sampling configuration. While this is done on purpose to not miss any drops, a side effect is that the reported proportion of drops versus non-drops becomes biased. For example, at a very high sampling rate, such as `1:1000`, it is likely that almost all the traffic appears to be dropped when observed from the console plugin. (link:https://issues.redhat.com/browse/NETOBSERV-1676[*NETOBSERV-1676*])

Reviewer 3: The jira RN text is still for to the 1.6.0 known issue, it should be updated such as:

```suggestion
* When the eBPF agent `PacketDrop` feature is enabled, and sampling is configured to a value greater than `1`, reported dropped bytes and dropped packets was ignoring the sampling configuration. While this was done on purpose to not miss any drops, a side effect was that the reported proportion of drops versus non-drops became biased. For example, at a very high sampling rate, such as `1:1000`, it was likely that almost all the traffic appears to be dropped when observed from the console plugin. The new behaviour is to honour the sampling configuration with dropped bytes and packets. (link:https://issues.redhat.com/browse/NETOBSERV-1676[*NETOBSERV-1676*])
```

Reviewer 3: (I also updated jira RN)

Author: Ok thank you. Addressed.

> * When the eBPF agent `PacketDrop` feature was enabled, and sampling was configured to a value greater than `1`, reported dropped bytes and dropped packets ignored the sampling configuration. While this was done on purpose, so as not to miss any drops, a side effect was that the reported proportion of drops versus non-drops became biased. For example, at a very high sampling rate, such as `1:1000`, it was likely that almost all the traffic appears to be dropped when observed from the console plugin. With this fix, the sampling configuration is honored with dropped bytes and packets. (link:https://issues.redhat.com/browse/NETOBSERV-1676[*NETOBSERV-1676*])

Reviewer 2: ISG says to spell versus out _or_ to use a similar phrasing, so this can be ignored I think

* Previously, the SR-IOV secondary interface was not detected if the interface was created first and then the eBPF agent was deployed. It was only detected if the agent was deployed first and then the SR-IOV interface was created. Now, the SR-IOV secondary interface is detected no matter the sequence of the deployments. (link:https://issues.redhat.com/browse/NETOBSERV-1697[*NETOBSERV-1697*])

> * Previously, when Loki was disabled, the *Topology* view in the OpenShift web console displayed the *Cluster* and *Zone* aggregation options in the slider beside the network topology diagram, even when the related features are not enabled. This has been fixed, the slider now only display options according to the enabled features. (link:https://issues.redhat.com/browse/NETOBSERV-1705[*NETOBSERV-1705*])

Reviewer 2: Should everything in the first sentence be in past tense?
```suggestion
* Previously, when Loki was disabled, the *Topology* view in the OpenShift web console displayed the *Cluster* and *Zone* aggregation options in the slider beside the network topology diagram, even when the related features were not enabled. This has been fixed, the slider now only display options according to the enabled features. (link:https://issues.redhat.com/browse/NETOBSERV-1705[*NETOBSERV-1705*])
```

Author: AH good catch yeah this was a known issue that became a bug fix, looks like holdover present tense.

> * Previously, when Loki was disabled, the *Topology* view in the OpenShift web console displayed the *Cluster* and *Zone* aggregation options in the slider beside the network topology diagram, even when the related features are not enabled. This has been fixed, the slider now only display options according to the enabled features. (link:https://issues.redhat.com/browse/NETOBSERV-1705[*NETOBSERV-1705*])

Reviewer 2: I think displays should end with an "s" since slider is singular:
```suggestion
* Previously, when Loki was disabled, the *Topology* view in the OpenShift web console displayed the *Cluster* and *Zone* aggregation options in the slider beside the network topology diagram, even when the related features are not enabled. This has been fixed, the slider now only displays options according to the enabled features. (link:https://issues.redhat.com/browse/NETOBSERV-1705[*NETOBSERV-1705*])
```

* Previously, when Loki was disabled, and the OpenShift web console was first loading, an error would occur: `Request failed with status code 400 Loki is disabled`. With this fix, the errors no longer occur. (link:https://issues.redhat.com/browse/NETOBSERV-1706[*NETOBSERV-1706*])

> * Previously, in the *Topology* view of the OpenShift web console, when clicking on the *Step into* icon next to any graph node, the filters were not applied as required in order to set the focus to the selected graph node, resulting in showing a wide view of the *Topology*. Now the filters are correctly set, effectively narrowing down the *Topology*. As part of this change, clicking the *Step into* icon on a graph node now brings you to *Resource* scope instead of *Namespaces* scope. (link:https://issues.redhat.com/browse/NETOBSERV-1720[*NETOBSERV-1720*])

Reviewer 3: here we really mean *Node* (the k8s Node), although the *graph node* is also technically correct.. in fact it's "the graph node of Note type" :-)
but I think just *Node* works better

```suggestion
* Previously, in the *Topology* view of the OpenShift web console, when clicking on the *Step into* icon next to any graph node, the filters were not applied as required in order to set the focus to the selected graph node, resulting in showing a wide view of the *Topology*. Now the filters are correctly set, effectively narrowing down the *Topology*. As part of this change, clicking the *Step into* icon on a *Node* now brings you to *Resource* scope instead of *Namespaces* scope. (link:https://issues.redhat.com/browse/NETOBSERV-1720[*NETOBSERV-1720*])
```

> * Previously, in the *Topology* view of the OpenShift web console, when clicking on the *Step into* icon next to any graph node, the filters were not applied as required in order to set the focus to the selected graph node, resulting in showing a wide view of the *Topology*. Now the filters are correctly set, effectively narrowing down the *Topology*. As part of this change, clicking the *Step into* icon on a *Node* now brings you to the *Resource* scope instead of the *Namespaces* scope. (link:https://issues.redhat.com/browse/NETOBSERV-1720[*NETOBSERV-1720*])

Reviewer 2: Nit - I feel like "resulting in showing a wide view of the *Topology*" is missing something. Perhaps it needs to specify the noun that is performing the "showing" action, or, since this is talking about a system, this might be an acceptable use case for passive voice per the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice):
```suggestion
* Previously, in the *Topology* view of the OpenShift web console, when clicking on the *Step into* icon next to any graph node, the filters were not applied as required in order to set the focus to the selected graph node, resulting in a wide view of the *Topology* being shown. Now the filters are correctly set, effectively narrowing down the *Topology*. As part of this change, clicking the *Step into* icon on a *Node* now brings you to the *Resource* scope instead of the *Namespaces* scope. (link:https://issues.redhat.com/browse/NETOBSERV-1720[*NETOBSERV-1720*])
```

Author: Yeah I think I need to say ...Topology view in OpenShift web console. Good one.

> * Previously, when Loki was disabled, in the *Topology* view of the OpenShift web console with the *Scope* set to *Owner*, clicking on the *Step into* icon next to any graph node would bring the *Scope* to *Resource*, which isn't available without Loki, so an error message was shown. Now the *Step into* icon is hidden in the *Owner* scope when Loki is disabled, so this scenario no longer occurs.(link:https://issues.redhat.com/browse/NETOBSERV-1721[*NETOBSERV-1721*])

Reviewer 2: I guess this contraction should be removed per [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#contractions):
```suggestion
* Previously, when Loki was disabled, in the *Topology* view of the OpenShift web console with the *Scope* set to *Owner*, clicking on the *Step into* icon next to any graph node would bring the *Scope* to *Resource*, which is not available without Loki, so an error message was shown. Now the *Step into* icon is hidden in the *Owner* scope when Loki is disabled, so this scenario no longer occurs.(link:https://issues.redhat.com/browse/NETOBSERV-1721[*NETOBSERV-1721*])
```

* Previously, when Loki was disabled, an error was displayed in the *Topology* view of the OpenShift web console when a group was set, but then the scope was changed so that the group becomes invalid. Now the invalid group is removed, preventing the error. (link:https://issues.redhat.com/browse/NETOBSERV-1722[*NETOBSERV-1722*])

> * When creating a `FlowCollector` resource from the OpenShift web console *Form view*, as opposed to the *YAML view*, the following settings are incorrectly managed by the web console: `agent.ebpf.metrics.enable` and `processor.subnetLabels.openShiftAutoDetect`. These settings can only be disabled in the *YAML view*, not in the *Form view*, . To avoid any confusion, these settings have been removed from the form view. They are still accessible in YAML. (link:https://issues.redhat.com/browse/NETOBSERV-1731[*NETOBSERV-1731*])

Reviewer 3: ```suggestion
* When creating a `FlowCollector` resource from the OpenShift web console *Form view*, as opposed to the *YAML view*, the following settings are incorrectly managed by the web console: `agent.ebpf.metrics.enable` and `processor.subnetLabels.openShiftAutoDetect`. These settings can only be disabled in the *YAML view*, not in the *Form view*. To avoid any confusion, these settings have been removed from the form view. They are still accessible in YAML. (link:https://issues.redhat.com/browse/NETOBSERV-1731[*NETOBSERV-1731*])
```

Author: Oof thanks 🥴

> * Previously, the eBPF agent was unable to clean up traffic control flows installed before an ungraceful crash, for example a crash due to a SIGTERM signal. This led to the creation of multiple traffic control flow filters with the same name, since the older ones were not removed. With this fix, all previously installed traffic control flows are cleaned up when the agent starts, before installing new ones. *NETOBSERV-1732*

Reviewer 2: This entry is missing a link to the bug

Author: Its bc this is a private jira, so I had to remove it.

Author: Just confirmed that this Jira can be set to a lower security level, as its already been customer-facing issue.
